src/Symfony/Component/Yaml/Tests/DumperTest.php

```diff
@@ -434,6 +435,98 @@ public function testDumpingStdClassInstancesRespectsInlineLevel()
     inner2: c
     inner3: { deep1: d, deep2: e }
 
+YAML;
+        $this->assertSame($expected, $yaml);
+    }
+
+    public function testDumpingTaggedValueSequenceRespectsInlineLevel()
+    {
+        $data = [
+            new TaggedValue('user', [
+                'username' => 'jane',
+            ]),
+            new TaggedValue('user', [
+                'username' => 'john',
+            ]),
+        ];
+
+        $yaml = $this->dumper->dump($data, 2);
+
+        $expected = <<<YAML
+- !user
+  username: jane
+- !user
+  username: john
+
+YAML;
+        $this->assertSame($expected, $yaml);
+    }
+
+    public function testDumpingTaggedValueSequenceWithInlinedTagValues()
+    {
+        $data = [
+            new TaggedValue('user', [
+                'username' => 'jane',
+            ]),
+            new TaggedValue('user', [
+                'username' => 'john',
+            ]),
+        ];
+
+        $yaml = $this->dumper->dump($data, 1);
+
+        $expected = <<<YAML
+- !user
+  { username: jane }
+- !user
+  { username: john }
+
+YAML;
+        $this->assertSame($expected, $yaml);
+    }
+
+    public function testDumpingTaggedValueMapRespectsInlineLevel()
+    {
+        $data = [
+            'user1' => new TaggedValue('user', [
+                'username' => 'jane',
+            ]),
+            'user2' => new TaggedValue('user', [
+                'username' => 'john',
+            ]),
+        ];
+
+        $yaml = $this->dumper->dump($data, 2);
+
+        $expected = <<<YAML
+user1: !user
+    username: jane
+user2: !user
+    username: john
+
+YAML;
+        $this->assertSame($expected, $yaml);
+    }
+
+    public function testDumpingTaggedValueMapWithInlinedTagValues()
+    {
+        $data = [
+            'user1' => new TaggedValue('user', [
+                'username' => 'jane',
+            ]),
+            'user2' => new TaggedValue('user', [
+                'username' => 'john',
+            ]),
+        ];
+
+        $yaml = $this->dumper->dump($data, 1);
+
+        $expected = <<<YAML
+user1: !user
```

Comment:
do we want to put inline object on the same line as their tag?

Reply:
sounds like a good idea

Reply:
done